### ISSUE TITLE
how to just train final layer

### ISSUE THREAD

**Issue by umichyiwan** [apache/incubator-mxnet#3555]
I would like to train final layer with previous layer fixed. Can I do something like this?
in symbol file

```
fc6_weight = mx.sym.Variable('fc6_weight', attr={'lr_mult':'0'})
fc6_bias = mx.sym.Variable('fc6_bias', attr={'lr_mult':'0'})
myfc7_weight = mx.sym.Variable('myfc7_weight')
myfc7_bias = mx.sym.Variable('myfc7_bias')
myfc7 = mx.symbol.FullyConnected(data=relu6, num_hidden=256, weight=myfc7_weight, bias=myfc7_bias)
myrelu7 = mx.symbol.Activation(data=myfc7, act_type="relu")
```

in the main function

```
model = mx.model.FeedForward(
            ctx=self.ctx,
            symbol=get_net(),
            num_epoch=self.num_epoch,
            learning_rate=0.001,
            momentum=0.9,
            wd=0.001,
            initializer=mx.init.Xavier(factor_type="in", magnitude=2.34),
            begin_epoch=begin_epoch,
)
```

Now what will the learning_rate be in layer 6 and layer 7?
There might be 3 cases:
layer 6 and 7 both are 0.001
layer 6 is 0 while layer 7 is 0.001
layer 6 is 0 while layer 7 is 0.01(default value)
Which one is correct?

**Comment by wangg12** [comment]
Is there any way to print out the whole network's architecture and the corresponding attributes including learning rate? It will be convenient for users to check if the defined architecture is expected.

**Comment by shuokay** [comment]
@wangg12 `mx.viz.plot_network` can visualize the network, but NO learning rate.

**Comment by shuokay** [comment]
@umichyiwan fc6 is 0 and myfc7 is 0.001.
If `lr_mult` is set, the learning rate used is `lr_mult*learning_rate`

**Comment by VoVAllen** [comment]
You can refer to the BlockGrad Operator to fine-tune the final layer

**Comment by VoVAllen** [comment]
If you want to set different learning rate for different layers, you need to update them manually with executor api

**Comment by starimpact** [comment]
this is easy to do: just split it into two nets. the first one is your pre trained layers. and the second one is your last layer. the input of the second one is the output of the first one. it is easy to implemented.

**Comment by yajiedesign** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
